lomond/parser.py

```diff
@@ -81,27 +105,34 @@ def feed(self, data):
         pos = 0
         while pos < len(data):
             # Awaiting a read of a fixed number of bytes
-            if isinstance(self._awaiting, self.read):
+            if isinstance(self._awaiting, _ReadBytes):
                 # This many bytes left to read
                 remaining = self._awaiting.remaining
                 # Bite off remaining bytes
                 chunk = data[pos:pos + remaining]
                 chunk_size = len(chunk)
                 pos += chunk_size
+                try:
```

Comment:
If the utf8 validation fails, it will throw a `ParseError`. But we want that exception to be thrown in the generator, so that it appears to come from the `yield` statement. Like magic it is.